l10n_ch_lsv_dd/wizard/lsv_export_wizard.py

```diff
@@ -0,0 +1,537 @@
+
+        for order in self.banking_export_ch_dd_id.payment_order_ids:
+            wf_service.trg_validate(self.env.uid, 'payment.order',
+                                    order.id, 'done', self.env.cr)
+            mandate_ids = list(set(
+                [line.mandate_id.id for line in order.line_ids]))
+            mandates = self.env['account.banking.mandate'].browse(mandate_ids)
+            mandates.write({'last_debit_date': today_str})
+
+        # redirect to generated lsv export
+        action = {
+            'name': 'Generated File',
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'form,tree',
+            'res_model': 'banking.export.ch.dd',
+            'res_id': self.banking_export_ch_dd_id.id,
+            'target': 'current',
+        }
+        return action
+
+    @api.multi
+    def cancel_export(self):
+        ''' Cancel the export: delete export record '''
+        self.banking_export_ch_dd_id.unlink()
+        return {'type': 'ir.actions.act_window_close'}
+
+    @api.model
+    def _customize_lines(self, lsv_lines, properties):
+        ''' Use this if you want to customize the generated lines.
+            @param lsv_lines: list of tuples with tup[0]=payment line
+                              and tup[1]=generated string.
+            @return: list of strings.
+        '''
+        return [tup[1] for tup in lsv_lines]
+
+    ##########################
+    #         Tools          #
+    ##########################
+    def _check_amount(self, line, properties):
+        ''' Max allowed amount is CHF 99'999'999.99.
+            We need to also check EUR values...
+        '''
+        if (properties.get('currency') == 'CHF' and
+            line.amount_currency > 99999999.99) or (
+                properties.get('currency') == 'EUR' and
+                line.amount_currency > 99999999.99 / properties.get('rate')):
+            raise exceptions.ValidationError(
+                _('Stop kidding... max authorized amount is CHF 99 999 999.99 '
+                  '(%.2f %s given for ref %s)') %
+                (line.amount_currency, properties.get('currency'), line.name))
+
+        elif line.amount_currency <= 0:
+            raise exceptions.ValidationError(
+                _('Amount for line with ref %s is negative (%f given)') %
+                (line.name, line.amount_currency))
+
+    def _check_currency(self, line, properties):
+        ''' Check that line currency is equal to lsv export currency '''
+        if not line.currency.name == properties.get(
+                'currency'):  # All currencies have to be the same !
+            raise exceptions.ValidationError(
+                _('Line with ref %s has %s currency and lsv file %s '
+                  '(should be the same)') %
+                (line.name, line.currency.name, properties.get(
+                    'currency', '')))
+
+    def _complete_line(self, string, nb_char):

```

Comment:
You use a lot string='', maybe set a default value by interverting string and nb_char to have string=''?